server/services/user.service.js

```diff
@@ -651,3 +651,35 @@ export const removeDirector = async ({ directorId, user }) => {
     throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error removing director', error);
   }
 };
+
+export const certifyVendor = async ({ vendorId, adminId }) => {
+  const user = await getUserById(vendorId);
+
+  if (!user) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'User not found');
+  }
+
+  if (user.role !== USER_ROLE.VENDOR) {
+    throw new ErrorHandler(httpStatus.PRECONDITION_FAILED, 'User is not a vendor');
+  }
+
+  if (user.vendor.verified !== true) {
+    throw new ErrorHandler(httpStatus.PRECONDITION_FAILED, 'Vendor has not been verified');
```

Comment:
Might want to use the company name here too to make the message more personalized.


```suggestion
      throw new ErrorHandler(httpStatus.PRECONDITION_FAILED, `${user.vendor.companyName} must be verified before approval as a certified vendor`);
```